src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -1529,5 +1530,218 @@ private bool TryGetNumberLookForEnd(ReadOnlySpan<byte> data, out ReadOnlySpan<by
             number = data.Slice(0, i);
             return true;
         }
+
+        public string GetValueAsString()
```

Comment:
Why don't we also have TryGetValueAsNNN alternatives that don't throw InvalidCastException?

Reply:
We already have the `Utf8Parser.TryParse` methods that the caller can use instead. These APIs are meant to be convenience APIs which throw if there is invalid data rather than returning a value based on partial consumption. For example, if the user calls `GetValueAsInt32` for a span containing a decimal, this throws. If they want to truncate instead, `Utf8Parser.TryParse` already does that.